ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala

```diff
@@ -59,31 +69,46 @@ object LedgerApiBenchTool {
           streamConfig.streamType match {
             case Config.StreamConfig.StreamType.Transactions =>
               TransactionMetrics
-                .transactionsMetricsManager(streamConfig.name, config.reportingPeriod)(system)
+                .transactionsMetricsManager(
+                  streamConfig.name,
+                  config.reportingPeriod,
+                  streamConfig.objectives,
+                )(system)
                 .flatMap { manager =>
                   val observer: MeteredStreamObserver[GetTransactionsResponse] =
                     new MeteredStreamObserver[GetTransactionsResponse](
                       streamConfig.name,
                       logger,
                       manager,
-                    )
+                    )(system)
                   transactionService.transactions(streamConfig, observer)
                 }
             case Config.StreamConfig.StreamType.TransactionTrees =>
               TransactionMetrics
-                .transactionTreesMetricsManager(streamConfig.name, config.reportingPeriod)(system)
+                .transactionTreesMetricsManager(
+                  streamConfig.name,
+                  config.reportingPeriod,
+                  streamConfig.objectives,
+                )(system)
                 .flatMap { manager =>
                   val observer =
                     new MeteredStreamObserver[GetTransactionTreesResponse](
                       streamConfig.name,
                       logger,
                       manager,
-                    )
+                    )(system)
                   transactionService.transactionTrees(streamConfig, observer)
                 }
           }
         }
-        .map(_ => ())
+        .transform {
+          case Success(results) =>
+            if (results.contains(MetricsManager.Message.MetricsResult.ObjectivesViolated))
+              Failure(new RuntimeException("Metrics objectives violated."))
```

Comment:
`violated` makes me thing about some criticality. How about `not met`?

Reply:
Are you fine with changing just error message and leaving changing variables/classes names for a follow-up PR?

Reply:
Sure